tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/domainmodels/endpoints/EndpointTestParamsDeserializer.java

```diff
@@ -57,7 +60,24 @@ public static EndpointTests.EndpointTestParameter deserializeParameter(String pa
                 // this will always be ["*"], however this should not be relied upon.
                 parameter.setType(EndpointTests.EndpointTestParameter.ParameterType.STRING);
                 parameter.setStrValue("*");
-            } else {
+            }
+            else if(element.isJsonArray())
```

Comment:
why is this nested in the else statement? shouldnt this be

```
if (element.isJsonPrimitive()) {
  ...
} else if (parameterName.equals("signingRegionSet") {
  ...
} else if(element.isJsonArray()) {
  ...
} else {
  ...
}
```

Reply:
The net effect is the same. I just added else if to the existing top level else block.

Reply:
```
if (element.isJsonPrimitive()) {
 ...
} else {
 if(element.isJsonArray()) {
   ...
  } else if (element.isJsonObject()) {
   ...
  } else {
   ...
  }
}
```
is a bit of anti pattern since they all operate on the same object they should be on the same level